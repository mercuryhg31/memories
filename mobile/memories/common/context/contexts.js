import { createContext, useContext, useState } from 'react';

const CurrentUserContext = createContext();
function CurrentUserContextProvider({ children }) {
  const [currentUserID, setCurrentUser] = useState('');
  const [displayUser, setDisplayUser] = useState(false);
  const [targetUserUID, setTargetUserUID] = useState('');
  const endpointURL = 'https://reqres.in';

  return (
    <CurrentUserContext.Provider
      value={{
        currentUserID,
        setCurrentUser,
        displayUser, // boolean of if the profile modal should be displayed
        setDisplayUser,
        targetUserUID,
        setTargetUserUID,
        endpointURL,
      }}>
      {children}
    </CurrentUserContext.Provider>
  );
};

export { CurrentUserContext, CurrentUserContextProvider };