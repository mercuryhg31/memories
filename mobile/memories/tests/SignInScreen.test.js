import React from 'react';
import renderer from 'react-test-renderer';
import { CurrentUserContextProvider } from '../common/context/contexts';
import SignInScreen from '../common/screens/SignInScreen';


describe('<SignInScreen/>', () => {
  test('has 1 child', () => {
    const tree = renderer.create(<CurrentUserContextProvider><SignInScreen/></CurrentUserContextProvider>).toJSON();
    expect(tree.children.length).toBe(1);
  });

  test('renders correctly', () => {
    const tree = renderer.create(<CurrentUserContextProvider><SignInScreen/></CurrentUserContextProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});