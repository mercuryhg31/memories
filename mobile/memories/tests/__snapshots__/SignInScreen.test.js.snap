// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInScreen/> renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#c5effc",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <RCTSafeAreaView>
        <View
          style={
            {
              "alignItems": "center",
              "paddingTop": 133.4,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://as1.ftcdn.net/v2/jpg/01/04/73/84/1000_F_104738464_uK24Z5qUllReax2nCDZ0HQiVINOf1k7m.jpg",
              }
            }
            style={
              {
                "borderRadius": 100,
                "height": 200.1,
                "width": 200.1,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "center",
                "top": 40.019999999999996,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 20,
                }
              }
            >
              Welcome to
            </Text>
            <Text
              style={
                {
                  "fontSize": 40,
                }
              }
            >
              Memories
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 40.019999999999996,
              "top": 66.7,
            }
          }
        >
          <View
            accessibilityLabel="Email input field"
            onLayout={[Function]}
            style={
              [
                {
                  "backgroundColor": "white",
                  "borderColor": "#e8e8e8",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                  "width": "80%",
                },
                {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="email"
              style={{}}
            />
          </View>
          <View
            accessibilityLabel="Password input field"
            onLayout={[Function]}
            style={
              [
                {
                  "backgroundColor": "white",
                  "borderColor": "#e8e8e8",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "height": 40,
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                  "width": "80%",
                },
                {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="password"
              style={{}}
            />
          </View>
          <View
            accessibilityLabel="Forgot password button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#858585",
                }
              }
            >
              Forgot Password
            </Text>
          </View>
          <View
            accessibilityLabel="Login button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#45d9a8",
                "borderColor": "#57ab8f",
                "borderRadius": 5,
                "borderWidth": 1,
                "opacity": 1,
                "paddingVertical": 10,
                "width": "80%",
              }
            }
          >
            <Text>
              Login
            </Text>
          </View>
          <View
            accessibilityLabel="Sign up button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#5ba1e3",
                "borderColor": "#57ab8f",
                "borderRadius": 5,
                "borderWidth": 1,
                "opacity": 1,
                "paddingVertical": 10,
                "width": "80%",
              }
            }
          >
            <Text>
              Sign Up
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </RCTScrollView>
</View>
`;
